### ISSUE TITLE
Optimizer state when changing MP(Model Parallelism) SIZE

### ISSUE THREAD

**Issue by SeonggwanAhn** [THUDM/GLM#29]
Can you tell me why [change_mp.py](https://github.com/THUDM/GLM/blob/main/change_mp.py#L41) does not preserve 'optimizer'??

Not preserving optimizer states matters me because there is a case where I need to change the number of GPUs during pretraining.

But current [change_mp.py](https://github.com/THUDM/GLM/blob/main/change_mp.py#L41) doesn't preserve 'optimizer states', so it makes error(Nonetype object is not subscriptable at [this line](https://github.com/THUDM/GLM/blob/main/utils.py#L371))) when I load GLM checkpoint after decrease MP_SIZE of current checkpoint.

Otherwise, is there any way to change the number of GPUs during pretraining?
\* I used the torch distributed package instead of DeepSpeed.

Thanks

**Comment by duzx16** [comment]
We didn't support changing the model parallelism size for optimizer states because it is difficult to implement in DeepSpeed's zero optimizer. A possible solution is not loading the optimizer state (#24) and warmup the learning rates from start.
